presto-main/src/main/java/com/facebook/presto/sql/planner/TimestampRuleChange.java

```diff
@@ -0,0 +1,144 @@
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.spi.function.StandardFunctionResolution;
+import com.facebook.presto.spi.plan.FilterNode;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.relational.FunctionResolution;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+import static com.facebook.presto.common.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static com.facebook.presto.common.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.sql.planner.plan.Patterns.filter;
+import static com.facebook.presto.sql.relational.Expressions.comparisonExpression;
+import static com.facebook.presto.sql.relational.Expressions.constant;
+
+/**
+ * Minimal iterative rule that rewrites:
+ *   CAST(ts_col AS date) = DATE 'yyyy-mm-dd'
+ * into
+ *   ts_col >= TIMESTAMP 'yyyy-mm-dd 00:00:00.000'
+ * (No upper bound to avoid referencing "AND".)
+ */
+public class TimestampRuleChange
```

Comment:
nit: let's be a little more specific with the name. How about `ConvertDateTimestampToTimestampBounds`